pkg/hub/addon/addon.go

```diff
@@ -1,66 +1,230 @@
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/open-cluster-management/cluster-proxy-addon/pkg/config"
+	"github.com/open-cluster-management/cluster-proxy-addon/pkg/helpers"
+	bindata "github.com/open-cluster-management/cluster-proxy-addon/pkg/hub/addon/bindata"
+	"github.com/openshift/library-go/pkg/assets"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/klog/v2"
 	"open-cluster-management.io/addon-framework/pkg/agent"
-	addonapiv1alpha1 "open-cluster-management.io/api/addon/v1alpha1"
+	"open-cluster-management.io/api/addon/v1alpha1"
 	clusterv1 "open-cluster-management.io/api/cluster/v1"
 
 	"github.com/openshift/library-go/pkg/operator/events"
 
 	certificatesv1 "k8s.io/api/certificates/v1"
 	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/client-go/kubernetes"
+)
+
+var (
+	genericScheme = runtime.NewScheme()
+	genericCodecs = serializer.NewCodecFactory(genericScheme)
+	genericCodec  = genericCodecs.UniversalDeserializer()
 )
 
 const (
-	addOnName = "cluster-proxy"
-	agentName = "cluster-proxy-agent"
+	addOnGroup         = "system:open-cluster-management:addon:cluster-proxy"
+	agentUserName      = "system:open-cluster-management:cluster:%s:addon:cluster-proxy:agent:cluster-proxy-addon-agent"
+	clusterAddOnGroup  = "system:open-cluster-management:cluster:%s:addon:cluster-proxy"
+	authenticatedGroup = "system:authenticated"
 )
 
+func init() {
+	scheme.AddToScheme(genericScheme)
+}
+
 type clusterProxyAddOnAgent struct {
-	kubeClient kubernetes.Interface
-	recorder   events.Recorder
-	agentImage string
+	recorder      events.Recorder
+	hubKubeClient *kubernetes.Clientset
+	agentName     string
+	agentImage    string
 }
 
 // NewClusterProxyAddOnAgent returns an instance of clusterProxyAddOnAgent
-func NewClusterProxyAddOnAgent(kubeClient kubernetes.Interface, recorder events.Recorder, agentImage string) *clusterProxyAddOnAgent {
+func NewClusterProxyAddOnAgent(hubKubeClient *kubernetes.Clientset, recorder events.Recorder, agentImage string) *clusterProxyAddOnAgent {
 	return &clusterProxyAddOnAgent{
-		kubeClient: kubeClient,
-		recorder:   recorder,
-		agentImage: agentImage,
+		recorder:      recorder,
+		hubKubeClient: hubKubeClient,
+		agentName:     config.ADDON_AGENT_NAME,
+		agentImage:    agentImage,
 	}
 }
 
 // Manifests generates manifestworks to deploy the clusternet-proxy-addon agent on the managed cluster
-func (a *clusterProxyAddOnAgent) Manifests(
-	cluster *clusterv1.ManagedCluster, addon *addonapiv1alpha1.ManagedClusterAddOn) ([]runtime.Object, error) {
-	//TODO
-	return nil, nil
+func (a *clusterProxyAddOnAgent) Manifests(cluster *clusterv1.ManagedCluster, addon *v1alpha1.ManagedClusterAddOn) ([]runtime.Object, error) {
+	installNamespace := addon.Spec.InstallNamespace
+	if len(installNamespace) == 0 {
+		installNamespace = config.AGENT_NAMESPACE
```

Comment:
if the install namespace is not specified, to keep the consistent with other acm addons, it is better to use `open-cluster-management-agent-addon`

Reply:
Done